eng/native/gen-buildsys.cmd

```diff
@@ -73,7 +74,7 @@ if /i "%__Arch%" == "wasm" (
         set "WASI_SDK_PATH=!WASI_SDK_PATH:\=/!"
         if not "!WASI_SDK_PATH:~-1!" == "/" set "WASI_SDK_PATH=!WASI_SDK_PATH!/"
         set __CmakeGenerator=Ninja
-        set __ExtraCmakeParams=%__ExtraCmakeParams% -DCLR_CMAKE_TARGET_OS=wasi -DCLR_CMAKE_TARGET_ARCH=wasm "-DWASI_SDK_PREFIX=!WASI_SDK_PATH!" "-DCMAKE_TOOLCHAIN_FILE=!WASI_SDK_PATH!share/cmake/wasi-sdk.cmake" "-DCMAKE_SYSROOT=!WASI_SDK_PATH!share/wasi-sysroot"
+        set __ExtraCmakeParams=%__ExtraCmakeParams% -DCLR_CMAKE_TARGET_OS=wasi -DCLR_CMAKE_TARGET_ARCH=wasm "-DWASI_SDK_PREFIX=!WASI_SDK_PATH!" "-DCMAKE_TOOLCHAIN_FILE=!__repoRoot!/src/native/external/wasi-sdk-p2.cmake" "-DCMAKE_SYSROOT=!WASI_SDK_PATH!share/wasi-sysroot" "-DCMAKE_CROSSCOMPILING_EMULATOR=!EMSDK_NODE! --experimental-wasm-bigint --experimental-wasi-unstable-preview1"
```

Comment:
I suspect this `EMSDK_NODE` reference is the last remaining dependency on Emscripten when building for WASI.  It seems fine for CI, where we install the Emscripten SDK anyway, but it would be nice to be able to build locally without installing it.

I guess folks could just set `EMSDK_NODE=node` in their environment, and we could document that as a legitimate option.  Perhaps we could also rename it to e.g. `NODE_EXECUTABLE` to make it clear that there's nothing Emscripten-specific about the version of Node required?

Reply:
`CMAKE_CROSSCOMPILING_EMULATOR` is actually a downstream-only need. We can:

1) Delete it here, and add downstream.
2) Delete it here, and hardcode the relevant cmake test results in `tryrun.cmake`.
2) Leave it here as just `node` (from `PATH`).

My slight preference is option `3`.

Reply:
the unix version also had just 3), updated